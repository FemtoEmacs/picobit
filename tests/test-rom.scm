;; let's see if this end up in rom, if so, vectors whose contents are known at compile time are likely to be optimized to rom for free
(define x 3)
(define y '(1 2 3)) ;; ok, if quote is used, goes to rom, if not, goes to ram, the compiler doesn't check to see if everything is known at compile-time, I probably should add this optimization
(set-car! y 9)
(display y)
