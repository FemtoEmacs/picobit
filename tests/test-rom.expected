:0080000080
:10800000FBD7200280050118811C4000811140002F
:10801000810D4000810B4000810C40008042000235
:10802000802701198027011A8024011B8071011DFE
:10803000806E011E800700028027011F80770120CB
:108040008073012180240002806A00028073012273
:1080500080670123800601108040011280400113D7
:10806000803201148027011580660116806901178E
:1080700080780002807D010E8066010F8070010A09
:108080008078010A02B401685002B4016251A1046F
:108090000DD0A104B582EED80120DAB8954121DBDC
:1080A000B6800221CC9A21CE2161EE2022CFB67279
:1080B00000EF20B6800120DA9EA02607E620B55802
:1080C000EEA02607E600EF20C0B8BF20B582B64874
:1080D000012004C8B8AEA03121B5A402B583CDDC1F
:1080E000EF02A034210EC5C122CD2102CDB400F192
:1080F0007101210EC89220EF210E406221B662016B
:108100000421C2EF2002B559B75520CCB8B8A02C35
:1081100007E620CEB51FEE20CFB6800120D296A074
:108120002D07E600EF20CC9DA02407E620CEB50564
:10813000EE20CFB666A107B00098EE20B000B5EEF5
:10814000A02D07E600EF20D995A108B1009820F0F6
:108150009C2095A106B10098A105B10098A109B194
:108160000098012007E600EF022004C8212302CD79
:10817000CDB4017571012204C822242302CDCDCDD6
:10818000B40184710123B8DC21B000FF2322260250
:10819000CDCDCDB40197710122B8C621B000FF2129
:1081A000242602CDCDCDB401AA71012022242602BD
:1081B000CDCDCDCDB401B870002121C4232502CD91
:1081C000CDB401C5710122B893219220EF2298010C
:1081D0009420B100FF20EF21B87BB77500B86EB7CF
:0781E0006A21B73B21B7251E
:00000001FF
